src/utils/parse.ts

```diff
@@ -63,12 +63,27 @@ export function secToDate(sec: BCS.Uint64) {
   return new Date(ms);
 }
 
-export function formatToFullType(coinType: string) {
-  const [
-    address,
-    module,
-    struct,
-  ] = coinType.split("::");
-  const shortAddr = address.startsWith("0x") ? address.slice(2) : address;
-  return `0x${shortAddr.padStart(ADDRESS_HEX_LENGTH, "0")}::${module}::${struct}`;
+const regExSimpleStruct = /(0x[a-fA-F0-9]+)(::[^:,\s]+::[^:,\s]+)/;
+const regExCompoundStruct = /(0x[a-fA-F0-9]+::[^:,\s]+::[^:,\s]+)/g;
```

Comment:
Does this work for all cases?
like: 
- `0x1234::mod::struct<0x1::xxh::vector<0x1::st::a>>`
- `0x1::a::b<0x1::a::b<vector<0x1::a::b>>,0x1::a::b<u8,u64>>`
- `vector<0x1::a::b<u8>>`

Reply:
Yes. added test cases into the unit test.